lib/http1.c

```diff
@@ -224,22 +233,21 @@ static int create_chunked_entity_reader(struct st_h2o_http1_conn_t *conn)
     reader->super.handle_incoming_entity = handle_chunked_entity_read;
     memset(&reader->decoder, 0, sizeof(reader->decoder));
     reader->decoder.consume_trailer = 1;
-    reader->prev_input_size = conn->_reqsize;
 
     return 0;
 }
 
 static void handle_content_length_entity_read(struct st_h2o_http1_conn_t *conn)
 {
+    int complete = 0;
     struct st_h2o_http1_content_length_entity_reader *reader = (void *)conn->_req_entity_reader;
 
-    /* wait until: reqsize == conn->_input.size */
-    if (conn->sock->input->size < conn->_reqsize)
+    if (conn->req._body.bytes_received + conn->sock->input->size >= reader->content_length)
+        complete = 1;
+    if (!complete && conn->sock->input->size == 0)
         return;
 
-    /* all input has arrived */
-    conn->req.entity = h2o_iovec_init(conn->sock->input->bytes + conn->_reqsize - reader->content_length, reader->content_length);
-    on_entity_read_complete(conn);
+    handle_one_body_fragment(conn, conn->sock->input->size, complete);
```

Comment:
Doesn't this process excessive amount of input as request body if the last input consists of the end of the request body _and the first few bytes of the next request_ (i.e. pipelining)?

Please correct me if I'm incorrect, and if we have a test covering such case.

Reply:
That's right, i've added a test in fb8e95b that shows the previous code was broken. This is fixed now.